js/src/Ice/Properties.js

```diff
@@ -135,13 +136,19 @@ export class Properties {
             throw new InitializationException("Attempt to set property with empty key");
         }
 
-        // Finds the corresponding Ice property if it exists. Also logs warnings for unknown Ice properties and
-        // case-insensitive Ice property prefix matches.
-        const prop = Properties.findIceProperty(key, true);
+        // Check if the property is in an Ice property prefix. If so, check that it's a valid property.
+        const propertyArray = Properties.findIcePropertyArray(key);
+        if (propertyArray) {
+            const prop = Properties.findProperty(key.substring(propertyArray.name.length + 1), propertyArray);
 
-        // If the property is deprecated, log a warning
-        if (prop !== null && prop.deprecated) {
-            getProcessLogger().warning("setting deprecated property: " + key);
+            if (!prop) {
```

Comment:
Same here, to keep with the current style for null checks.

```suggestion
            if (prop === null) {
```